inst/include/child_model_simulation.hpp

```diff
@@ -731,16 +1422,37 @@ void run_child_model_simulation(int time_step,
                                 const State<ModelVariant, real_type> &state_curr,
                                 State<ModelVariant, real_type> &state_next,
                                 internal::IntermediateData<ModelVariant, real_type> &intermediate) {
+  const auto cpars = pars.children.children;
+
   internal::run_child_ageing(time_step, pars, state_curr, state_next, intermediate);
-  internal::run_child_hiv_infections(time_step, pars, state_curr, state_next, intermediate);
-  internal::run_child_natural_history(time_step, pars, state_curr, state_next, intermediate);
-  internal::run_child_hiv_mort(time_step, pars, state_curr, state_next, intermediate);
-  internal::add_child_grad(time_step, pars, state_curr, state_next, intermediate);
-  // this function may need to be broken up, its around 350 lines
-  // !!!TODO: also need to fix the looping order for some loops
-  // !!!TODO: put this in an if statement to only run if the first year of ART has passed
-  internal::run_child_art_initiation(time_step, pars, state_curr, state_next, intermediate);
-  internal::run_child_art_mortality(time_step, pars, state_curr, state_next, intermediate);
+
+  if(cpars.mat_prev_input(time_step)){
+    internal::run_wlhiv_births_input_mat_prev(time_step, pars, state_curr, state_next, intermediate);
+  }else{
+    internal::run_wlhiv_births(time_step, pars, state_curr, state_next, intermediate);
+  }
```

Comment:
Nit picky style comment
```suggestion
  if (cpars.mat_prev_input(time_step)) {
    internal::run_wlhiv_births_input_mat_prev(time_step, pars, state_curr, state_next, intermediate);
  } else {
    internal::run_wlhiv_births(time_step, pars, state_curr, state_next, intermediate);
  }
```

Reply:
Changed in all instances.